### DISCUSSION TITLE
Add Workflow to PNG

### DISCUSSION THREAD

**Discussion by WASasquatch** [Comfy-Org/ComfyUI#349] [Show and tell]
Add any workflow to any arbitrary PNG with this simple tool: https://rebrand.ly/workflow2png

Open the file browser and upload your images and json files, then simply copy their links (right click -> copy path) and paste them into the corresponding fields and run the cell. 

Also put together a quick CLI tool to use local. 

![image](https://user-images.githubusercontent.com/1151589/229226905-53a04fd3-69c2-419f-b0e6-4d029ad6c8d7.png)

**Comment by Buumcode** [comment]
This script is working for ComfyUI – final image successfully load as workflow; but this 'reconstructed' workflow (and prompt) not recognized in Civitai.com or https://i210i.github.io/
So, this is 'incorrect' workflow insertion. We need better version.
Here is comparison of 'good' native ComfyUI file and "bad" reconstructed via script file – made by https://www.nayuki.io/page/png-file-chunk-inspector
https://www.imagebam.com/view/MEXPOBO 
We need 'first section' of tEXt data, starting with 'Keyword: prompt'.

**Reply by Buumcode** [reply]
I've make script that allow inject 'prompt' and 'workflow' data back in PNG file.
Workflow_and_prompt_to_PNG -- https://disk.yandex.ru/d/Zr9_2USunypcMg
https://civitai.com/articles/9260
